fili-core/src/test/groovy/com/yahoo/bard/webservice/web/util/CacheServiceSpec.groovy

```diff
@@ -0,0 +1,203 @@
+import com.yahoo.bard.webservice.metadata.SegmentIntervalsHashIdGenerator
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+import com.yahoo.bard.webservice.web.RequestUtils
+import com.yahoo.bard.webservice.web.handlers.RequestContext
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseContext
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor
+import com.yahoo.bard.webservice.web.responseprocessors.WeightCheckResponseProcessor
+import org.joda.time.Interval
+import spock.lang.Shared
+import spock.lang.Specification
+
+import com.fasterxml.jackson.databind.JsonNode
+
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.core.MultivaluedHashMap
+
+import static com.yahoo.bard.webservice.async.ResponseContextUtils.createResponseContext
+import static com.yahoo.bard.webservice.web.responseprocessors.ResponseContextKeys.MISSING_INTERVALS_CONTEXT_KEY
+import static com.yahoo.bard.webservice.web.responseprocessors.ResponseContextKeys.VOLATILE_INTERVALS_CONTEXT_KEY
+
+class CacheServiceSpec extends Specification {
+    private static final ObjectMapper MAPPER = new ObjectMappersSuite().getMapper()
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.instance
+
+    ResponseProcessor response
+    TupleDataCache<String, Long, String> dataCache
+    CacheService cacheService
+    BardQueryInfo bardQueryInfo
+    JsonNode json
+
+    GroupByQuery groupByQuery
+    QuerySigningService querySigningService
+    ContainerRequestContext containerRequestContext
+    RequestContext requestContext
+
+    @Shared SimplifiedIntervalList intervals = new SimplifiedIntervalList()
+    @Shared SimplifiedIntervalList nonEmptyIntervals = new SimplifiedIntervalList([new Interval(0, 1)])
+    ResponseContext responseContext =  createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): intervals])
+
+
+    def setup() {
+        response = Mock(WeightCheckResponseProcessor)
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        bardQueryInfo = BardQueryInfoUtils.initializeBardQueryInfo()
+        dataCache = Mock(TupleDataCache)
+        json = new JsonNodeFactory().arrayNode()
+        groupByQuery = RequestUtils.buildGroupByQuery()
+        querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        querySigningService.getSegmentSetId(_) >> Optional.of(1234L)
+        containerRequestContext = Mock(ContainerRequestContext)
+        containerRequestContext.getHeaders() >> (["Bard-Testing": "###BYPASS###", "ClientId": "UI"] as
+                MultivaluedHashMap<String, String>)
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        MAPPER.valueToTree(groupByQuery) >> json
+    }
+
+    def "Test if request is cachable"() {
```

Comment:
I would just that you use 
1) @Unroll annotation
2) Use variable substitution in the test name #foo 
3) add a description column to the where clause, so that it reads something like:
```suggestion
    def "if #description then is cachable is #expected"() {
```
```suggestion
where:
expected | context | description
false    | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals]) | "missing intervals is non empty"
true    | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): new SimplifiedIntervalList()]) | "missing intervals is present and empty"
```